cfme/tests/infrastructure/test_host.py

```diff
@@ -426,3 +426,28 @@ def test_infrastructure_hosts_navigation_after_download(
     hosts_view.navigation.select("Compute")
     provider_view = provider.create_view(InfraProviderDetailsView)
     assert provider_view.is_displayed
+
+
+@test_requirements.infra_hosts
+@pytest.mark.meta(blockers=[BZ(1746214, forced_streams=["5.10"])], automates=[1746214])
+def test_compare_hosts_from_provider_allhosts(appliance, setup_provider_min_hosts, provider):
+    """
+    Polarion:
+        assignee: prichard
+        casecomponent: Infra
+        caseimportance: high
+        initialEstimate: 1/6h
+    Bugzilla:
+        1746214
+
+    """
+    hosts_view = navigate_to(provider.collections.hosts, "All")
+    num_hosts = len(hosts_view.entities.get_all())
```

Comment:
```suggestion
    hosts = hosts_view.entities.get_all()
    if len(hosts) < 2:
        pytest.skip('not enough hosts in appliance UI to run test')
    for h in hosts:
        h.check()
```

Reply:
There were a few updates to how I do this. I now use entities.paginator.items_amount